### ISSUE TITLE
`double_ended_iterator_last` should not trigger on immutable non-ref receiver

### ISSUE THREAD

**Issue by samueltardieu** [rust-lang/rust-clippy#14139]
### Summary

See title

### Lint Name

double_ended_iterator_last

### Reproducer

```rust
fn main() {
    let mut index = [true,true,false,false,false,true].iter();
    let subindex = index.by_ref().take(3);
    subindex.last();
}
```

suggests using `subindex.next_back()` which won't work because `subindex` cannot be auto-derefed into a `ref mut`.

### Version

```text
rustc 1.86.0-nightly (2f348cb7c 2025-01-27)
binary: rustc
commit-hash: 2f348cb7ce4063fa4eb40038e6ada3c5214717bd
commit-date: 2025-01-27
host: x86_64-unknown-linux-gnu
release: 1.86.0-nightly
LLVM version: 19.1.7
```

**Comment by samueltardieu** [comment]
Signaled in <https://github.com/rust-lang/rust-clippy/pull/13922#issuecomment-2629433822>
